components/style/values/specified/length.rs

```diff
@@ -576,7 +576,6 @@ impl CalcLengthOrPercentage {
         let mut ch = None;
         let mut rem = None;
         let mut percentage = None;
-        let mut number = None;
```

Comment:
Firefox didn't get `calc()` right for properties which support numbers (particular `line-height`), but people are fixing that.

Not returning number as part of the result but treating it valid make no sense anyway. For properties which doesn't accept numbers, it is pretty consistent to reject `calc()` contains top level numbers. You can try this testcase:
```html
<!DOCTYPE html>
<style>
div {
  width: 1px;
  width: calc(5); /* invalid */
  width: calc(10 + 20px); /* invalid */
}
</style>
<script>
document.write(document.styleSheets[0].cssRules[0].style.width);
</script>
```

It would likely crash Servo because of serializing empty `calc()`. See the stack backtrace in [bug 1321206](https://bugzilla.mozilla.org/show_bug.cgi?id=1321206).